src/spatialdata/_core/query/relational_query.py

```diff
@@ -147,17 +150,83 @@ def _filter_table_by_elements(
+    table_instance_key_column: pd.Series,
+    match_rows: Literal["left", "right"],
+) -> pd.Index:
+    """
+    Get indices of the table that are present in element_indices.
+
+    Parameters
+    ----------
+    joined_indices
+        Current indices that have been found to match indices of an element
+    element_indices
+        Element indices to match against table_instance_key_column.
+    table_instance_key_column
+        The column of a table containing the instance ids.
+    match_rows
+        Whether to match the indices of the element and table and if so how. If left, element_indices take priority and
+        if right table instance ids take priority.
+
+    Returns
+    -------
+        The indices that of the table that match the SpatialElement indices.
+    """
+    mask = table_instance_key_column.isin(element_indices)
+    if joined_indices is None:
+        if match_rows == "left":
+            joined_indices = _match_rows(table_instance_key_column, mask, element_indices, match_rows)
+        else:
+            joined_indices = table_instance_key_column[mask].index
+    else:
+        if match_rows == "left":
+            add_indices = _match_rows(table_instance_key_column, mask, element_indices, match_rows)
+            joined_indices = joined_indices.append(add_indices)
+        # in place append does not work with pd.Index
+        else:
+            joined_indices = joined_indices.append(table_instance_key_column[mask].index)
+    return joined_indices
+
+
+def _get_masked_element(
+    element_indices: pd.RangeIndex,
+    element: SpatialElement,
+    table_instance_key_column: pd.Series,
+    match_rows: Literal["left", "right"],
+) -> SpatialElement:
+    """
+    Get element rows matching the instance ids in the table_instance_key_column.
+
+    Parameters
+    ----------
+    element_indices
+        The indices of an element.
+    element
+        The spatial element to be masked.
+    table_instance_key_column
+        The column of a table containing the instance ids
+    match_rows : Literal["left", "right"]
+         Whether to match the indices of the element and table and if so how. If left, element_indices take priority and
+        if right table instance ids take priority.
+
+    Returns
+    -------
+    The masked spatial element based on the provided indices and match rows.
+    """
+    mask = table_instance_key_column.isin(element_indices)
+    masked_table_instance_key_column = table_instance_key_column[mask]
+    mask_values = mask_values if len(mask_values := masked_table_instance_key_column.values) != 0 else None
+    if match_rows == "right":
+        mask_values = _match_rows(table_instance_key_column, mask, element_indices, match_rows)
+
+    return element.loc[mask_values, :]
 
 
 def _right_exclusive_join_spatialelement_table(
-    element_dict: dict[str, dict[str, Any]], table: AnnData
+    element_dict: dict[str, dict[str, Any]], table: AnnData, match_rows: str
```

Comment:
I'd replace `str`, for consistency, with `why str and not `Literal['no' ,'right']`

Reply:
Also a check of the valid directions could be added.

Reply:
will add the check later. You are right